### PR TITLE
Update vendor dependencies

### PR DESCRIPTION
**Pull request by humblec**
Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

```release-note
Updated client-go to v0.18
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -6,16 +6,16 @@ require (
...
 	github.com/davecgh/go-spew v1.1.1
 	github.com/evanphx/json-patch v4.5.0+incompatible
-	github.com/golang/mock v1.2.0
-	github.com/golang/protobuf v1.3.2
-	github.com/imdario/mergo v0.3.7 // indirect
+	github.com/golang/mock v1.4.3
+	github.com/golang/protobuf v1.3.5
+	github.com/imdario/mergo v0.3.9 // indirect
 	github.com/kubernetes-csi/csi-lib-utils v0.7.0
 	github.com/kubernetes-csi/csi-test v2.0.0+incompatible
```
Comment (pohly) [inline go.mod:13]: Can you also update to csi-test v3.x?

**Comment by humblec** [comment]
@jsafrane @pohly looks like CI is passing now, I am removing `WIP` . However I have added `ctx` and other params to API call to satisfy the new func signature due to the dependency update. Please review ..

File: go.mod (hunk 1 above)
Comment (humblec) [reply go.mod]: Done..thx.

**Comment by msau42** [comment]
Yes, let's update all our repos to 1.18 client-go

**Comment by humblec** [comment]
https://github.com/kubernetes-sigs/sig-storage-lib-external-provisioner/pull/73 cover sslexternal provisioner.

File: pkg/controller/util.go [hunk 2]
```diff
@@ -26,11 +27,14 @@ import (
 	jsonpatch "github.com/evanphx/json-patch"
 	v1 "k8s.io/api/core/v1"
 	storage "k8s.io/api/storage/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/klog"
 )
 
+var ctx context.Context = context.Background()
```
Comment (jsafrane) [inline pkg/controller/util.go:36]: Please do not use global variables. Use `context.TODO()` in all places where you use this.
Or even better, find a good API server timeout and use that.

Comment (humblec) [reply pkg/controller/util.go:36]: Sure, let me start with context.TODO() local variable. Will do a follow up PR for configurable timeout . thanks !

Comment (humblec) [reply pkg/controller/util.go]: @jsafrane done.. changed to context.TODO() . .. @msau42 @pohly please review!

**Comment by msau42** [comment]
/lgtm
/approve

This may be worth adding a release note since the client-go update was signficant. Please add one and then unhold
/hold

**Comment by humblec** [comment]
@msau42 @pohly @jsafrane added release note . Please verify and unhold :).. thanks!

**Comment by jsafrane** [comment]
The release note is for end users (admins), who do not care about internals of golang API calls. IMO something like "Updated client-go to v0.18" would be sufficient.

**Comment by humblec** [comment]
> The release note is for end users (admins), who do not care about internals of golang API calls. IMO something like "Updated client-go to v0.18" would be sufficient.

@Sure.. done.. !

**Comment by jsafrane** [comment]
Fixing failed travis by closing and reopening...

**Comment by humblec** [comment]
I was wondering why this was stuck.. thanks @jsafrane :+1:

**Comment by msau42** [comment]
travis is still stuck :( going to force merge. everything else is passing fine.

**Comment by humblec** [comment]
Thanks @jsafrane and @msau42 !!

**Pull request merged**